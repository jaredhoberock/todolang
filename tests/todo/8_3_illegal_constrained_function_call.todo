fun foo<T, U: Add>(y: T, x: U) -> Number {
  0
}

// this should fail because a constraint couldn't be solved
foo(true, true);
