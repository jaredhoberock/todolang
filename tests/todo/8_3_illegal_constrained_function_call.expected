[0m[1m[38;5;9merror[0m[1m: The trait bound 'Bool: Add' is not satisfied'[0m
  [0m[34mâ”Œâ”€[0m tests/todo/8_3_illegal_constrained_function_call.todo:6:11
  [0m[34mâ”‚[0m
[0m[34m6[0m [0m[34mâ”‚[0m foo(true, [0m[31mtrue[0m);
  [0m[34mâ”‚[0m [0m[34m---[0m       [0m[31m^^^^[0m
  [0m[34mâ”‚[0m [0m[34mâ”‚[0m         [0m[34mâ”‚[0m
  [0m[34mâ”‚[0m [0m[34mâ”‚[0m         [0m[34mthe trait 'Add' is not implemented for 'Bool'[0m
  [0m[34mâ”‚[0m [0m[34mrequired by a bound introduced by this call[0m


